src/cluster.c

```diff
@@ -5922,44 +6115,127 @@ NULL
-                 * configEpoch collision resolution will fix it assigning
-                 * a different epoch to each node. */
-                if (clusterBumpConfigEpochWithoutConsensus() == C_OK) {
-                    serverLog(LL_NOTICE,
-                        "configEpoch updated after importing slot %d", slot);
-                }
                 server.cluster->importing_slots_from[slot] = NULL;
-                /* After importing this slot, let the other nodes know as
-                 * soon as possible. */
-                clusterBroadcastPong(CLUSTER_BROADCAST_ALL);
+
+                /* Only primary broadcasts the updates */
+                if (n == myself) {
+                    /* This slot was manually migrated, set this node configEpoch
+                    * to a new epoch so that the new version can be propagated
+                    * by the cluster.
+                    *
+                    * Note that if this ever results in a collision with another
+                    * node getting the same configEpoch, for example because a
+                    * failover happens at the same time we close the slot, the
+                    * configEpoch collision resolution will fix it assigning
+                    * a different epoch to each node. */
+                    if (clusterBumpConfigEpochWithoutConsensus() == C_OK) {
+                        serverLog(LL_NOTICE,
+                                  "ConfigEpoch updated after importing slot %d",
+                                  slot);
+                    }
+                    /* After importing this slot, let the other nodes know as
+                    * soon as possible. */
+                    clusterBroadcastPong(CLUSTER_BROADCAST_ALL);
+                }
+            }
+        } else if (!strcasecmp(c->argv[3]->ptr,"node") && c->argc == 6 && !strcasecmp(c->argv[5]->ptr,"replicaonly")) {
+            /* CLUSTER SETSLOT <SLOT> NODE <NODE ID> REPLICAONLY */
+
+            /* When finalizing the slot, there is a possibility that the
+             * target node B sends a cluster PONG to the source node A
+             * before SETSLOT has been replicated to B'. If B crashes here,
+             * B' will be in an importing state and the slot will have no
+             * owner. To help mitigate this issue, we added a new SETSLOT
+             * command variant that takes a special marker token called
+             * "REPLICAONLY". This command is a no-op on the primary. It
+             * simply replicates asynchronously the command without the
+             * "REPLICAONLY" marker to the replicas, if there exist any.
+             * The caller is expected to wait for this asynchronous
+             * replication to complete using the "WAIT" command.
+             *
+             * With the help of this command, we finalizes the slots
+             * on the replicas before the primary in the following
+             * sequence:
```

Comment:
Explaining what A and B are before the sequence instead of afterwards makes it easier to follow the sequence IMO.

```suggestion
             * sequence, where A is the source primary and B is the target
             * primary:
```